cdc/generation.cc

```diff
@@ -776,7 +776,6 @@ generation_service::~generation_service() {
 
 future<> generation_service::after_join(std::optional<cdc::generation_id>&& startup_gen_id) {
     assert_shard_zero(__PRETTY_FUNCTION__);
-    assert(_sys_ks.local().bootstrap_complete());
```

Comment:
Well the point of such sanity checks is precisely to check preconditions like the one you described in the commit message. You could argue that all assertions are "unneeded" because they check preconditions that must always be true, so they are true... as long as there's no bug in the code :)

Reply:
Well, yes I agree :) but

The ultimate plan is to move cached bootstrap state from sysks cache onto storage service local `join_cluster` variable (or maybe a class member), because it's not used other than through this method execution. And this CDC place stands on the way :(

Reply:
OK, let's remove it then.